Modules/TPC/src/TrendingTaskTPC.cxx

```diff
@@ -0,0 +1,292 @@
+                } else {
+                  ILOG(Error, Devel) << "Could not get the title 'TPaveText' of the plot '"
+                    << plot.name << "'." << ENDM;
+                }*/
+        /// LOKI
+
+        // Set the user-defined range on the y axis if needed.
+        if (!plot.graphYRange.empty()) {
+          std::size_t posDivider = plot.graphYRange.find(":");
+          std::string yMinString(plot.graphYRange.substr(0, posDivider));
+          std::string yMaxString(plot.graphYRange.substr(posDivider + 1));
+
+          float yMin = std::stof(yMinString);
+          float yMax = std::stof(yMaxString);
+          histo->SetMinimum(yMin);
+          histo->SetMaximum(yMax);
+        }
+
+        // Configure the time for the x axis.
+        if (plot.varexp.find(":time") != std::string::npos) {
+          histo->GetXaxis()->SetTimeDisplay(1);
+          histo->GetXaxis()->SetNdivisions(505);
+          histo->GetXaxis()->SetTimeOffset(0.0);
+          histo->GetXaxis()->SetTimeFormat("%Y-%m-%d %H:%M");
+        }
+
+        // Manually empty the buffers before visualising the plot.
+        // histo->BufferEmpty(); // TBD: Should we keep it or not?
+      } else {
+        ILOG(Error, Devel) << "Could not get the 'Graph' of the plot '"
+                           << plot.name << "'." << ENDM;
+      }
+    }
+
+    mPlots[plot.name] = c;
+    getObjectsManager()->startPublishing(c);
+  }
+}
+
+void TrendingTaskTPC::drawCanvas(TCanvas* thisCanvas, const std::string& var,
+                                 const std::string& sel, const std::string& opt, const std::string& err,
+                                 const std::string& name)
+{
+  // Divide the canvas into the correct number of pads.
+  thisCanvas->DivideSquare(mNumberPads);
+
+  // Determine the order of the plot (1 - histo, 2 - graph, ...)
+  const size_t plotOrder = std::count(var.begin(), var.end(), ':') + 1;
+
+  // Delete the graph errors after the plot is saved.
+  // Unfortunately the canvas does not take its ownership.
+  TGraph* graphPad = nullptr;
+  TGraphErrors* graphErrors = nullptr;
+
+  // Prepare the strings for the dataSource and its trending quantity.
+  std::size_t posEndVar = var.find(".");  // Find the end of the dataSource.
+  std::size_t posEndType = var.find(":"); // Find the end of the quantity.
+  std::string varName(var.substr(0, posEndVar));
+  std::string typeName(var.substr(posEndVar + 1, posEndType - posEndVar - 1));
+
+  std::size_t posEndType_err = err.find(":"); // Find the end of the error.
+  std::string errXName(err.substr(posEndType_err + 1));
+  std::string errYName(err.substr(0, posEndType_err));
+
+  // Setup the tree reader with the needed values.
+  TTreeReader myReader(mTrend.get());
+  TTreeReaderValue<UInt_t> RetrieveTime(myReader, "time");
+  TTreeReaderValue<std::vector<SliceInfo>> DataRetrieveVector(myReader, varName.data());
+
+  int iEvent = 0;
+  const int nEvents = mTrend->GetEntriesFast();
```

Comment:
"Event" is maybe not the best choice for the number of tree entries (i.e. time stamps), maybe better "iEntry", "nEntries"?